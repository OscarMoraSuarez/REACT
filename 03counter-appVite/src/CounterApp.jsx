import { useState }from 'react';
import { PropTypes } from "prop-types";

//?cunado la funcion no ocupa nada en el componente en el cual se aloja
//? se recomienda declararla y usarla afuera



const CounterApp=({value})=>{

    const [ counter,setCounter ]=useState(value)
    
    const sumar = () => {

        // setCounter(counter+1);
        setCounter(counter+1)

    }
     const restar = () =>  setCounter(counter - 1);

    
    const resetear = () => setCounter( value) ;

    
   

return(

    <>
        <h1> CounterApp </h1>
        <h2 data-testid="counter"> { counter } </h2>
        <button aria-label='btn-sumar' onClick={ sumar }> 
        +1
        </button>
        <button aria-label='btn-restar' onClick={ restar }> 
        -1 
        </button >
        <button id='btn-reset' onClick={ resetear }> 
        reset
        </button>
    </>

)
    
    




}

export {

    CounterApp

}


CounterApp.propTypes={

    value:PropTypes.number.isRequired


}

/* CounterApp.defaultProps={

    value:100, 

}
 */
    