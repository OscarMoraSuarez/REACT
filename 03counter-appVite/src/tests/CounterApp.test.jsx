import { fireEvent, render,screen } from "@testing-library/react"
import { CounterApp } from "../CounterApp"



describe('Pruebas en comp  <CounterApp/> ', () => { 
    const number=10;
    test('debe hacer juego con el snapshot', () => { 
        const {container}=render(<CounterApp value={number}/>);
        expect(container).toMatchSnapshot();
       
    });
      test('el contenido del h2 debe ser "100"',     () => { 
        
         render(<CounterApp value={100} />);
        /* screen es la representacion del componente que etenemos renderizado */
          expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100'); 
          /* expect(screen.getByTestId('counter').innerHTML).toContain(100);  */
          /* expect(screen.getByText('100')).toBeTruthy(); */
      });

      test('Debe incrementar con el boton +1', () => { 

            render(<CounterApp value={number}/>);
            fireEvent.click(screen.getByText('+1')); 
            
            expect(screen.getByText('11')).toBeTruthy();


       });
       test('Debe decrementar con el boton -1', () => { 

           render(<CounterApp value={number}/> )
           fireEvent.click(screen.getByText('-1')); 
           
           expect(screen.getByText('9')).toBeTruthy();
        });

        test('Debe funcionar el botÃ²n de reset', () => { 

            render(<CounterApp value={355} />)
            fireEvent.click(screen.getByText('+1')); 
            fireEvent.click(screen.getByText('+1')); 
            fireEvent.click(screen.getByText('+1')); 
            /* fireEvent.click(screen.getByText('reset')); */
            /* screen.debug(); */
            fireEvent.click(screen.getByRole('button',{name:'reset'}));
             expect(screen.getByText(355)).toBeTruthy(); 





         })
})