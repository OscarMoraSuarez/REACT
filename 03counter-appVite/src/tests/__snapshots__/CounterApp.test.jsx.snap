// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en comp  <CounterApp/>  debe hacer juego con el snapshot 1`] = `
<div>
  <h1>
     CounterApp 
  </h1>
  <h2
    data-testid="counter"
  >
     
    10
     
  </h2>
  <button
    aria-label="btn-sumar"
  >
    +1
  </button>
  <button
    aria-label="btn-restar"
  >
    -1
  </button>
  <button
    id="btn-reset"
  >
    reset
  </button>
</div>
`;
